src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -221,14 +221,22 @@ trait Implicits {
     }
 
     def isCyclicOrErroneous: Boolean = {
-      if (!isCyclicOrErroneousCache.isKnown) isCyclicOrErroneousCache = computeIsCyclicOrErroneous
-      isCyclicOrErroneousCache.booleanValue
+      if (!isCyclicOrErroneousCache.isKnown)
+        try global.withPropagateCyclicReferences {
+          val cyclicOrErroneous = sym.hasFlag(LOCKED) || containsError(tpe)
+          isCyclicOrErroneousCache = cyclicOrErroneous
+          cyclicOrErroneous
+        } catch {
+          case _: CyclicReference =>
```

Comment:
There's a lot of noise on the duration of tests. We do have benchmarking infra coming up soon /cc @retronym